### ISSUE TITLE
About the arc from 'test.fits'

### ISSUE THREAD

**Issue by WentaoLuo** [linan7788626/Arc_Finding_with_CMU#2]
Is there a lens galaxy or  the smoothed profile of the dark matter that creates the  arcs?

**Comment by linan7788626** [comment]
yes, the mass model of this lens is Non-singular Isothermal Ellipsoid, and the velocity dispersion is 320 km/s.

**Comment by linan7788626** [comment]
@WentaoLuo try this http://www.syntevo.com/smartgit/ for using github under linux.

**Comment by WentaoLuo** [comment]
Thanks, I am fine with git now after one night training.  It is basically SSH.

**Comment by WentaoLuo** [comment]
I am wondering if we could combine the arc image and simulated SDSS image together and form a
more realistic one?

**Comment by WentaoLuo** [comment]
And I would then simulate more image, maybe 10000. And add others to th github.

**Comment by linan7788626** [comment]
hold on, I have question, is sdssgal is the image with noise? you told me the noise should be white noise, but in this fits, it is not flat...

**Comment by WentaoLuo** [comment]
It is Gaussian.
With a galaxy image  in the center.

**Comment by WentaoLuo** [comment]
And also notice that the image size is not 120x120, instead. It is 119x119, or things like that due to 
the scaling from cosmos to SDSS by SHERA.

**Comment by WentaoLuo** [comment]
When do you think we add other people?

**Comment by linan7788626** [comment]
How do you identify the parameters of the gaussian distribution? it should be related to the dark matter halo we are using. Also, please set the images size to be 128x128. it should be easier for both of us. 

> When do you think we add other people?
> At least, after producing one simulated images. Do you have any suggestion?

**Comment by WentaoLuo** [comment]
I will go to some details later. Sure, 128x128 sounds good.

**Comment by linan7788626** [comment]
for the lensed images using toy models, you do not need to worry about the image of the lens galaxy, please just give noise and PSF.

**Comment by WentaoLuo** [comment]
The Gaussian noise is 58.7 in terms of variance (in unit of count^2, photon-electron counts), and the mean is 0.0.

**Comment by WentaoLuo** [comment]
I will update the sdss galaxy image soon with 128x128.

**Comment by linan7788626** [comment]
the PSF you give me is not normalized......

**Comment by WentaoLuo** [comment]
You mean the size is not 128x128?

**Comment by WentaoLuo** [comment]
Or I could try combine those images together?

**Comment by WentaoLuo** [comment]
The convolution between them in IDL does not require the same size for both images.

**Comment by linan7788626** [comment]
no, I mean the summary of all pixels is not one.

**Comment by linan7788626** [comment]
the values of the pixels in the sdssgal.fits is too high, what is the units of the value of these pixels?
![screen shot 2015-12-08 at 1 46 07 pm](https://cloud.githubusercontent.com/assets/1016652/11666446/242cc918-9db2-11e5-899e-14fde696394f.png)

**Comment by WentaoLuo** [comment]
Both is  galaxy image and PSF image there are softbias=1000.0, before any scale. the images should subtract 1000.0. For the psf,  after subtraction, scale the image  using psf/total(psf).

**Comment by linan7788626** [comment]
all right, before we combine them together, the most important thing is that we have to make sure we are using the same unit in all images.....

**Comment by linan7788626** [comment]
I updated a new lensed image which is in ./output_fits/, it looks realistic, but the flux of source and lens  are  incorrect. And another issue, noise should be added separately, but not added with lens galaxy. otherwise, how could we convolve PSF for the lens galaxy. Anyway, please tell me what is the unit of the value of the pixels in the fits files you gave me.

**Comment by WentaoLuo** [comment]
what I use is count of photon-electron, produced by CCD. It mimics SDSS detector. 
The count magnitude relation is :http://classic.sdss.org/dr7/algorithms/fluxcal.html.
f/f0 = counts/exptime \* 10^{-0.4_(aa + kk \* airmass)}, And
number of photon-electron=count_gain, gain_r=4.7
mag = -2.5 \* log10(f/f0)
SDSS exposure time 54 minutes.
We simply set {-0.4_(aa+kk_airmass)) to be -9.6 meaning we fix a zero-point and airmass.

Then, given an wanted SDSS magnitude MAG
MAG--->f/f0--->counts.

By the way, I think the unit we use should be counts rather than photon-electron, so we do not need gain_r value.

Sorry for that.

**Comment by WentaoLuo** [comment]
![figure_1](https://cloud.githubusercontent.com/assets/6350744/11667118/e8059666-9dbd-11e5-8dc2-fc469768b74c.png)

Nice work, man!

**Comment by WentaoLuo** [comment]
We should definitely add the others to this project, so we could brainstorm.
What do you think. 
I think we are ready to simulate a larger set of images, don't we?

**Comment by linan7788626** [comment]
I know we should add others to this project, how about later this week, before that we have a couple of things to fix:
- The image of lens galaxy without noise.
- Calibrating the magnitudes of source galaxies and lens galaxies. 
- Modeling the surface brightness of lens galaxies according to the velocity dispersion of lenses.

If we do not figure our these problems, the images we produced are not naturally right. 
After this, we add them in the collaborators list. 
Does it make sense ?

**Comment by WentaoLuo** [comment]
Yes, you are right.
We should at least fix the above three issues before we add them.

**Comment by WentaoLuo** [comment]
I believe the velocity dispersion you need of the lens galaxy should be taken as the dispersion of the dark matter halo with an SIE profile?

**Comment by linan7788626** [comment]
the velocity dispersion I am using here is the velocity dispersion of stars in the lens galaxy. It is also the velocity dispersion of the total mass distribution, not dark matter halo only.

**Comment by WentaoLuo** [comment]
Or you mean the "the velocity dispersion of the lenses" are the dispersion of the dark matter halo?
And then using Faber-Jackson relation, we could estimate the luminosity of the galaxy reside in the halo?

**Comment by WentaoLuo** [comment]
I see.

**Comment by linan7788626** [comment]
do not be struggling in dark matter halo or what, let’s use total mass distribution here.

**Comment by WentaoLuo** [comment]
I am on it. velocity dispersion----->luminosity(Faber-Jackson relation)----> SDSS CCD counts.

**Comment by WentaoLuo** [comment]
Alright, I will add the according calibration python code to our repository.

**Comment by linan7788626** [comment]
how you model the surface brightness distribution? use 1/4 law please , if you can.

**Comment by WentaoLuo** [comment]
Sure, n=1/4. typical elliptical galaxy profile, a.k.a de Vaucouleurs profile.

**Comment by linan7788626** [comment]
yes

**Comment by WentaoLuo** [comment]
I hope your computer will be fine. I decide to make generate the lens galaxy profile using the  following
steps:
1, Fundamental plane: given a velocity dispersion(no scatter first)--> R_e, miu_e, luminosity
2, de Vaucouleurs profile with certain R_e
*3, We finally need to take the scatter of the fundamental plane into consideration, with assigning 
     a random Number from Normal distribution with the same scatter of a certain scaling relation, e.g.
     miu_e vs R_e, or vel_dis vs miu_e, e.t.c.

The attached is an example of fundamental plane,  

/home/wtluo/Desktop/figure2.jpg

**Comment by WentaoLuo** [comment]
![figure2](https://cloud.githubusercontent.com/assets/6350744/11717684/fe91bc44-9f1f-11e5-99e2-351158d8498e.jpg)

**Comment by linan7788626** [comment]
it looks great, everything sounds reasonable, but I have one question: how we identify the parameter I_0 of the de Vaucouleurs profile profile? we need to use I_0 and R_e to describe the profile fully.

**Comment by WentaoLuo** [comment]
Do not worry, in that profile,   the I_0 is actually I_e. People build the fundamental plane to make 
thins easier rather than more complex. See the link :https://en.wikipedia.org/wiki/De_Vaucouleurs'_law.
The only thing is the empirical number: 7.669, which change the I_0 to I_e. 

Once we have the fundamental plane, everything is good, except for the scatter.

**Comment by linan7788626** [comment]
Great, I am looking forward to your result of the lens galaxies, please do not forget the ellipticity and orientation, if possible just set them as input parameters as well.  here e = 0.8 , phi = 45 degree.

**Comment by WentaoLuo** [comment]
Sorry, I_o is not I_e. But can be scaled to I_e with some weird number 7.669.

**Comment by WentaoLuo** [comment]
Sure, I will take ellipticity into consideration.

**Comment by linan7788626** [comment]
could you Point me to the reference for identify I_e R_e and I_0 , please. Thanks

**Comment by WentaoLuo** [comment]
[AS3011_5.pdf](https://github.com/linan7788626/Arc_Finding_with_CMU/files/58406/AS3011_5.pdf)

I_e=I_0exp(-7.669)

**Comment by WentaoLuo** [comment]
We need to find out a paper that describe this relation.
The PDF is only from a lecture.

**Comment by WentaoLuo** [comment]
An example of Poisson noise
![poisson_noise](https://cloud.githubusercontent.com/assets/6350744/11752681/e90a6f16-a00e-11e5-96fb-bbef59d945fb.png)

**Comment by WentaoLuo** [comment]
An example of Gaussian noise.
![gaussian_noise](https://cloud.githubusercontent.com/assets/6350744/11752698/f969a534-a00e-11e5-8ed5-0bc361f0c0ab.png)

**Comment by WentaoLuo** [comment]
I will upload the python code soon. it is based on Rachel's work SHERA.

**Comment by WentaoLuo** [comment]
The simulated Gaussian has bigger dispersion.

**Comment by WentaoLuo** [comment]
These noise map are simulated with noise variance=58.75, typical SDSS noise variance.

**Comment by WentaoLuo** [comment]
The unit is count^2.

**Comment by WentaoLuo** [comment]
The poisson part may be wrong. 
I will check and modify it later.

**Comment by linan7788626** [comment]
Hi @WentaoLuo , I just merged your SDSS images , and will be working on it soon. thanks a lot.

**Comment by WentaoLuo** [comment]
I will focus on galaxy next.

**Comment by linan7788626** [comment]
so you think the noise model is right?

**Comment by WentaoLuo** [comment]
Yes, I talked with Rachel. it is right.

**Comment by linan7788626** [comment]
Hi wentao, I have merged your code and update the script for producing lensed images.  If the noise model is correct, then, so far, the only thing we need is the images of lenses, and please update your commits of the code in the same way. Thanks. BTW, Good luck to you about your paper. :-)

**Comment by WentaoLuo** [comment]
![figure_2](https://cloud.githubusercontent.com/assets/6350744/11853656/437244b8-a40d-11e5-929c-f5cc72c1168a.png)

Simulated de Vaucouleurs with R_e=2.9'', miu_e=27.989 z=0.3 with gaussian noise

**Comment by WentaoLuo** [comment]
![figure_3](https://cloud.githubusercontent.com/assets/6350744/11853716/9c19cb5e-a40d-11e5-97a1-163af495e74a.png)

This one is a figure without noise in ln(image) plot.  It seems it is too big. Maybe we have to add sky level rather than noise only.

**Comment by WentaoLuo** [comment]
./lens_galaxy.py  --E 0.5 --Phi 45 --R_e 2.0 --zl 0.1 --NoiseVar 59

Once I put the lens_galaxy.py to this repo, everyone can simulate an de Vaucouleurs image by adding parameters like ellipticity --E, position angle --Phi or effective radius --R_e e.t.c

**Comment by WentaoLuo** [comment]
The velocity dispersion of a galaxy with R_e and miu_e, can be calculated also in lens_galaxy.py.

**Comment by WentaoLuo** [comment]
![figure_4](https://cloud.githubusercontent.com/assets/6350744/11854111/b9aee4cc-a40f-11e5-86ff-60eedaaf2be4.png)

Yes, we definitely need to add sky to mimic SDSS survey.

**Comment by WentaoLuo** [comment]
How much information we lost from the sky!!!

**Comment by WentaoLuo** [comment]
This galaxy has miu_e~ 18.8825

**Comment by WentaoLuo** [comment]
I mean effective brightness

**Comment by linan7788626** [comment]
it looks great, I will get into it later. thanks. did you make a pull request? if so I will merge it asap.

**Comment by WentaoLuo** [comment]
![figure_5](https://cloud.githubusercontent.com/assets/6350744/11854509/50fc0538-a412-11e5-894a-6b7101010946.png)

this one is with Poisson noise. I think Guassian is better.

**Comment by WentaoLuo** [comment]
I will modify it and check 
1, the fundamental plane relation
2, noise modelling

And then put it on git and send a pull request.

**Comment by linan7788626** [comment]
OK, thanks, I will read your new scripts later. By the way, I did not see any noise in the latest figure. is the scales of noise and flux right?

**Comment by WentaoLuo** [comment]
The Poisson noise model has much smaller variance than Gaussian one, that is the reason.
I use exactly the model by Rachel. For now, I think we should use Gaussian, because it seems more
realistic.

**Comment by linan7788626** [comment]
can I ask why the poisson noise is much smaller than Gaussian? Is it realistic?

**Comment by linan7788626** [comment]
Hi Wentao, Could you please make a pull request for your new code ? so that I can merge them in the master branch. Thanks

**Comment by WentaoLuo** [comment]
I did a major modification of the simulation code.
The image magnitude will be given by velocity dispersion, effective radius(arcsec), redshift.
The noise from Rachel works well after this modification.

I will show you some example with fix redshift z=0.5, effetive radius=10,
and velocity dispersion=100,200,300

**Comment by WentaoLuo** [comment]
![vd_100](https://cloud.githubusercontent.com/assets/6350744/11902180/f28a3fde-a57d-11e5-8927-b9f3d9f810dc.png)
![vd_200](https://cloud.githubusercontent.com/assets/6350744/11902182/f3d466bc-a57d-11e5-87c5-37ebed13208d.png)
![vd_300](https://cloud.githubusercontent.com/assets/6350744/11902183/f51b9504-a57d-11e5-9ea4-b3fd00dfb3b3.png)

**Comment by WentaoLuo** [comment]
Next, we fix vd=300, radius_eff=10,
z=0.2,0.3,0.5

**Comment by WentaoLuo** [comment]
![z_0 2](https://cloud.githubusercontent.com/assets/6350744/11902261/8445103e-a57e-11e5-8720-05f8c2ba1f68.png)
![z_0 3](https://cloud.githubusercontent.com/assets/6350744/11902263/853f9702-a57e-11e5-923c-41aa858c1db1.png)
![z_0 5](https://cloud.githubusercontent.com/assets/6350744/11902264/86344338-a57e-11e5-8df9-1df0d4104872.png)

**Comment by WentaoLuo** [comment]
Sorry the size is fixed

**Comment by WentaoLuo** [comment]
Let's fix zl=0.3, vd=300 and 
R_eff=5,8,10

**Comment by WentaoLuo** [comment]
![reff_5](https://cloud.githubusercontent.com/assets/6350744/11902333/0b6282ea-a57f-11e5-9601-653bc121ea07.png)
![reff_8](https://cloud.githubusercontent.com/assets/6350744/11902335/0c807772-a57f-11e5-8ee0-699c56418852.png)
![reff_10](https://cloud.githubusercontent.com/assets/6350744/11902337/0e2f0a84-a57f-11e5-8b9b-bc450da88460.png)

**Comment by WentaoLuo** [comment]
I will push the code soon.

**Comment by WentaoLuo** [comment]
$./lens_galaxy.py --NoiseVar 59 --Phi 45 --E 0.3 --R_eff 8 --zl 0.3 --VelDis 300

The basic command is like that.
I will write a simple readme and then push the code and file soon after I finish the readme.

**Comment by WentaoLuo** [comment]
After this modification, the difference between the two noise models are not as huge as before.
There was a bug when calculating the counts of SDSS detectors.

**Comment by WentaoLuo** [comment]
![gaussian_noise_model](https://cloud.githubusercontent.com/assets/6350744/11906151/98050826-a598-11e5-931c-423468aa9ac9.png)
![poisson_noise_model](https://cloud.githubusercontent.com/assets/6350744/11906155/9f697c8c-a598-11e5-8789-75d06ea6d51e.png)

**Comment by WentaoLuo** [comment]
![gaussian_noise_model](https://cloud.githubusercontent.com/assets/6350744/11906180/bdc58c70-a598-11e5-9aeb-9a3c7031d1fa.png)

**Comment by WentaoLuo** [comment]
![guassian_noise_model](https://cloud.githubusercontent.com/assets/6350744/11906191/cbc531fe-a598-11e5-95a8-bc6285483d73.png)

**Comment by WentaoLuo** [comment]
![poisson_noise_model](https://cloud.githubusercontent.com/assets/6350744/11906196/d25a086e-a598-11e5-9ceb-7b892643ec5b.png)

**Comment by WentaoLuo** [comment]
Poisson has smaller standard deviation than Gaussian

**Comment by linan7788626** [comment]
Oh, it looks great, this is the noise we need, thanks a lot.

**Comment by WentaoLuo** [comment]
Debugging is tiring. Haha

**Comment by WentaoLuo** [comment]
Larger velocity dispersion, the brighter. If we set vd=300 km/s. The noise will be not as obvious as we set vd=150 or 100.

**Comment by linan7788626** [comment]
haha, BTW, how about the bug of your 2d kappa map? If needed, we can talk on it later.

**Comment by WentaoLuo** [comment]
Sure, I am focusing on this big bug. Later I will take care of that one. I could not focus at two things at the same time. Haha

**Comment by WentaoLuo** [comment]
![color_no_noise](https://cloud.githubusercontent.com/assets/6350744/12056593/13714896-af06-11e5-90a3-e5965351a514.png)

I think we should further to modify our pipeline a little bit to generate rgb color images of galaxy. like the above. This one is without noise. We should follow Lupton et al  2004 MNRAS 116 to do this. I will work on it.

**Comment by linan7788626** [comment]
Hi Wentao, 

It looks great, I am still on my way back to Chicago, I will working on this during the weekend. thanks a lot.
Happy new year! :)

-Nan

**Comment by linan7788626** [comment]
Hi Wentao,

How could I convert the unit of flux of the lens galaxy from number counts to magnitude properly?
Say, I have a 2d map as shown in the figure your attached, there should be some basic relation between number counts and magnitude, right?

Thanks,
Nan

**Comment by WentaoLuo** [comment]
![cosmos_example_1](https://cloud.githubusercontent.com/assets/6350744/12397800/712db56a-bddd-11e5-906e-5f4e2e2e7057.png)

**Comment by linan7788626** [comment]
thanks wentao, it looks great, BTW, what is the angular resolution of this image?

**Comment by WentaoLuo** [comment]
This is an example image from acs at z~1.0.
Sorry for the late update, I took my kid to get vaccine last Thurday and Friday was my older daughter's birthday.

**Comment by WentaoLuo** [comment]
Angular resolution interms of arcseconds?

**Comment by WentaoLuo** [comment]
I will check, give me a sec.

**Comment by linan7788626** [comment]
No worries, happy birthday to your girl, :-) 

> On Jan 18, 2016, at 11:19 AM, Wentao Luo notifications@github.com wrote:
> 
> Angular resolution interms of arc seconds?
> 
> Yes, I am wondering what the pixeliztions of this image is. Is it 0.03 ?

BTW, my cell phone is broken, so please contact me via Skype or email, if needed , thank you so much!

**Comment by WentaoLuo** [comment]
Yes, it is 0.03".
There are three different radius:
1,FLUX_RADIUS 29.1319 half light radius from SexTractor
2,KRON_RADIUS  6.32  Elliptical fit radius from SexTractor
3,PETRO_RADIUS  10.56  Petrosian fit radius 

All in arcseconds

**Comment by WentaoLuo** [comment]
Sorry for your loss.
I will contact you by skype or email then.

**Comment by WentaoLuo** [comment]
I think we could use the PETRO_RADIUS first.

**Comment by linan7788626** [comment]
PETRO_RADIUS for what? for lens galaxy?

**Comment by WentaoLuo** [comment]
For the size of source galaxy.

**Comment by linan7788626** [comment]
We already have the real image, so the size of source galaxy is not necessary. what I need the is the angular size of the pixel size of the real image.

**Comment by WentaoLuo** [comment]
I see. it is 0.03"/pix

**Comment by linan7788626** [comment]
Great, I am working on it now. will keep you updated.

**Comment by WentaoLuo** [comment]
Great. Thanks.

**Comment by linan7788626** [comment]
Hi Wentao, enclosed figure shows the simulated observation using the galaxy from Cosmos, it looks OK. 
I have one question is that did you push the converting function for different units? I did not find it. If possible, shall we discuss in tomorrow morning ? Thanks. 

![figure_3](https://cloud.githubusercontent.com/assets/1016652/12468646/07ed3902-bfab-11e5-9a97-fefa9007f002.png)

**Comment by WentaoLuo** [comment]
Hi Nan, It looks beautiful to me!  
Sure, let's talk today, 9pm for me and 8 pm for you.
How about that.

**Comment by WentaoLuo** [comment]
![figure_1](https://cloud.githubusercontent.com/assets/6350744/12565105/eb80dfec-c380-11e5-86eb-eb84afa19337.png)

Image with pixel value converted to SDSS pixel value without noise.

**Comment by WentaoLuo** [comment]
![figure_2](https://cloud.githubusercontent.com/assets/6350744/12565126/fd11f9bc-c380-11e5-90b0-b4218fe0a32b.png)

Image with pixel value converted to SDSS pixel value with Gaussian noise, Variance=58.0 in unit of DN(Data Number) from SDSS CCD.

**Comment by WentaoLuo** [comment]
![figure_3](https://cloud.githubusercontent.com/assets/6350744/12567109/44d33036-c38b-11e5-9ff6-f2be0ddba827.png)
![figure_4](https://cloud.githubusercontent.com/assets/6350744/12567111/4615a884-c38b-11e5-8013-39476891bd83.png)

Something is going wrong here. It seems the pixel value from Leauthaud  et al 2011 is not described 
as ACS paper, which is  in unit of electron/seconds. i will check and correct this from pixacs2pixsdss.py from image directory.

**Comment by WentaoLuo** [comment]
The above image with noise, i multiply the image by a factor of 200 to  make the galaxy  profile observable.

**Comment by linan7788626** [comment]
Hi all, 

Enclosed figure is an example of simulated lensed image, the only thing we need to do is making the flux of the source (from cosmos) and the flux of the lens (simulated SDSS image) matching to each other. Wentao and I are working on it. Once it is done, I will produce 1000 such images for testing, then we can play with those simulated images using machine learning pipelines. I will keep you all updated.

Thanks,
Nan

![figure_3](https://cloud.githubusercontent.com/assets/1016652/12649861/04dcca60-c5a6-11e5-9e93-a148b331375a.png)

**Comment by linan7788626** [comment]
Done. We know how to produce the image of lens using analytical models based on the fundamental plane of elliptical galaxies.
